pkg/apis/openebs.io/lvm/v1alpha1/lvmnode.go

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright © 2021 The OpenEBS Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=lvmnode
+
+// LVMNode records information about all lvm volume groups available
+// in a node. In general, the openebs node-agent creates the LVMNode
+// object & periodically synchronizing the volume groups available in the node.
+// LVMNode has an owner reference pointing to the corresponding node object.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,shortName=lvmnode
+type LVMNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	VolumeGroups []VolumeGroup `json:"volumeGroups"`
```

Comment:
Shouldn't the `VolumeGroups` be in the `Spec` section as per the convention in k8s API resources.

Reply:
In general yes, but not always. `Spec` is used to record the intent of resource & `Status` specify the current state of world (as seen by controllers). Here the `LVMNode` crd is informative rather than any spec (user intend per se) where controller tries to match it with external state. See [api-conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) for more details. There are many crds which doesn't follow spec/status pattern unless required. For example, in this context see [CSIStorageCapacity crd](https://github.com/kubernetes/kubernetes/blob/6d5cb36d36f34cb4f5735b6adcd5ea8ebb4440ba/staging/src/k8s.io/api/storage/v1alpha1/types.go#L160) which is again informative rather than a user intend.

Reply:
Are we not going to have the `Status` for  `lvmNode` resource ? IMO  then  it make sense to not have `Spec` Section